pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -152,3 +167,130 @@ func (c configResource) computeAndReconcileConfigMap(xsite *configuration.XSite)
+	r := newInfinispanOperatorConfigReconciler(mgr)
+	// Create a new controller
+	c, err := controller.New("infinispan-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	operatorNS, err := getOperatorNamespace()
+	if err != nil && err != k8sutil.ErrRunLocal {
+		return err
+	}
+	operatorConfigPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetName() == "infinispan-operator-config" && e.MetaNew.GetNamespace() == operatorNS
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+	}
+	// Watch for changes to global operator config
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, operatorConfigPredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// newInfinispanOperatorConfigReconciler returns a new reconcile.Reconciler
+func newInfinispanOperatorConfigReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileInfinispanOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme(), Log: logf.Log.WithName("operator-config-controller"), Kube: kube.NewKubernetesFromController(mgr)}
+}
+
+var _ reconcile.Reconciler = &ReconcileInfinispanOperatorConfig{}
+
+// ReconcileInfinispan reconciles a Infinispan object
+type ReconcileInfinispanOperatorConfig struct {
+	client client.Client
+	scheme *runtime.Scheme
+	Log    logr.Logger
+	Kube   *kube.Kubernetes
+}
+
+// Reconcile reads the operator configuration in the `infinispan-operator-config` ConfigMap and
+// applies the configuration to the cluster.
+func (r *ReconcileInfinispanOperatorConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	err := r.reconcileOperatorConfigMap()
+	if err != nil {
+		r.Log.Error(err, "Error while computing and reconciling ConfigMap")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// reconcileOperatorConfigMap reconcile the operator ConfigMap
+// Operator configMap is unique for an operator install and placed in k8sutil.GetOperatorNamespace()/infinispan-operator-config
+func (r ReconcileInfinispanOperatorConfig) reconcileOperatorConfigMap() error {
+	config := &corev1.ConfigMap{}
+	operatorNs, err := getOperatorNamespace()
+	isRunningLocal := false
+	if err == k8sutil.ErrRunLocal {
+		isRunningLocal = true
+	}
+	if err != nil && !isRunningLocal {
+		return err
+	}
+	r.client.Get(ctx, types.NamespacedName{Namespace: operatorNs, Name: "infinispan-operator-config"}, config)
+	// Reconcile grafana
+	grafanaNs := config.Data["grafanaDashboardNameSpace"]
```

Comment:
I think we should keep the grafana CRD creation in `pkg/infinispan/grafana/dashboard_controller.go` and call it from this reconcile loop. Adding all of the resource creation logic to the configuration reconcilation logic does not scale as we add additional configuration options.

Reply:
I would go this way:
-  split the config controller in a separate file
-  and move grafana utils in a separate file but in the same package, I don't think we need to have them global

Reply:
> * and move grafana utils in a separate file but in the same package, I don't think we need to have them global

Just to be clear. Are you proposing moving just the CR reconcilation to another file?

I think there's merit in keeping `pkg/infinispan/grafana/` for the implementation parts like the .json etc.

Reply:
I've moved grafana under the controller package since implementation is only used there